deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.kt

```diff
@@ -197,32 +238,180 @@ class DeepLinkProcessor(symbolProcessorEnvironment: SymbolProcessorEnvironment?
+                message = "For args constructor elements of type ${DeepLinkParamType.Query.name} only the following simple types are allowed: ${allowedNullableTypes.joinToString()}"
+            )
+            return
+        }
+        val deepLinkUriTemplate = DeepLinkUri.parseTemplate(uriTemplate)
+        val templateHostPathSchemePlaceholders = deepLinkUriTemplate.schemeHostPathPlaceholders
+        val annotatedPathParameterNames = allPathParameters.mapNotNull {
+            it.getAnnotation(DeeplinkParam::class)?.value?.name
+        }.toSet()
+        if (annotatedPathParameterNames != templateHostPathSchemePlaceholders) {
+            logError(
+                element = argsType.typeElement ?: element,
+                message = "All scheme/host/path placeholders in the uri template must be annotated in the argument class constructor. " +
+                    "Present in urlTemplate: ${templateHostPathSchemePlaceholders.joinToString()} " +
+                    "Present in constructor: ${annotatedPathParameterNames.joinToString()}"
+            )
+        }
+        val templateQueryParameters = deepLinkUriTemplate.queryParameterNames()
+        val annotatedQueryParameterNames = allQueryParameters.mapNotNull {
+            it.getAnnotation(DeeplinkParam::class)?.value?.name
+        }.toSet()
+        if (annotatedQueryParameterNames != templateQueryParameters) {
+            logError(
+                element = argsType.typeElement ?: element,
+                message = "All query elements in the uri template must be annotated in the argument class constructor. " +
+                    "Present in urlTemplate: ${templateQueryParameters.joinToString()} " +
+                    "Present in constructor: ${annotatedQueryParameterNames.joinToString()}"
+            )
+        }
+    }
+
+    private fun verifyObjectElement(element: XTypeElement): Boolean {
+        if (!element.isHandlerElement()) {
+            logError(
+                element = element,
+                message = "Only objects extending ${com.airbnb.deeplinkdispatch.handler.DeepLinkHandler::class.java.canonicalName} " +
+                    "can be annotated with @${DEEP_LINK_CLASS.simpleName}"
+            )
+            return false
+        }
+        if (element.getAllMethods()
+            .filter { it.name == DEEP_LINK_HANDLER_METHOD_NAME && it.parameters.size == 1 }.size != 1
         ) {
             logError(
-                element = methodElement,
-                message = (
-                    "Only `Intent`, `androidx.core.app.TaskStackBuilder` or " +
-                        "'com.airbnb.deeplinkdispatch.DeepLinkMethodResult' are supported. Please double " +
-                        "check your imports and try again."
-                    )
+                element = element,
+                message = "More than one method with a single parameter and $DEEP_LINK_HANDLER_METHOD_NAME name found in handler class."
```

Comment:
This should be impossible, right?

Reply:
@BenSchwab No, see the unit test testing that(1testDeeplinkHandlerWithTwoSingleParameterMethodsWithCorrectName`). It can have a different parameter type. And as the generics get deleted during runtime I don't know which of them would be the correct type if there are multiple.